### ISSUE TITLE
truncation json message in trace log

### ISSUE THREAD

**Issue by plvnv** [dimitri/pgcopydb#362]
Build pgcopydb with PR
```fix/resume-streaming-from-json```

```
10058 INFO   Running pgcopydb version 0.12.6.g5bc4aa1.dirty from "/opt/pgpro/std-13/bin/pgcopydb"
pgcopydb version 0.12.6.g5bc4aa1.dirty
compiled with PostgreSQL 13.10 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44), 64-bit
compatible with Postgres 10, 11, 12, 13, 14, and 15
```

Failed to parse JSON message:

[pgcopydb_trace.log](https://github.com/dimitri/pgcopydb/files/11923391/pgcopydb_trace.log)

```
{"action":"U","xid":"297905783","lsn":"2241/CA7C9C08","timestamp":"2023-06-30 21:20:37.969214+0300"
```

JSON message in [pgcopydb_trace.log](https://github.com/dimitri/pgcopydb/files/11923391/pgcopydb_trace.log) is truncated.

We can see the full request in the log  ```0000000200002241000000CA.json.partial```

**Comment by arajkumar** [comment]
Hello @dimitri 

I'm not sure it is relevant, I see a JSON parse error with `pgcopydb follow` command.

17:49:39 31449 ERROR  Failed to parse JSON message: {"action":"I","xid":"1280381","lsn":"26/F0021748","timestamp":"2023-07-04 12:19:38.478612+0530","message":{"action":"I","xid":1280381,"schema":"public","table":"r
17:49:39 31449 ERROR  Failed to transform JSON messages from input stream, see above for details
17:49:39 30923 ERROR  Subprocess transform with pid 31449 has exited with error code 12

The same is seen with `pgcopydb stream prefetch`, `pgcopydb stream receive --to-stdout | pgcopydb stream transform - -` as well.

IIUC, `pgcopydb receive` sends partial json message to the transform which is causing the trouble.

**Comment by dimitri** [comment]
Hi @arajkumar ; it looks like something else to me. First in your case it seems you are having an issue when using catchup mode, and the issue here is only happening in replay mode with Unix PIPEs. Also, the issue you have looks like it may have been solved in 0.12 thanks to better quoting of the values: in other words it could be a data format issue rather than a buffer size issue. Please try with a build from the current source code and report back with using `--trace` log level, and open a new issue.
